nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/validation/DocumentValidationException.java

```diff
@@ -38,10 +43,21 @@
     private DocumentValidationReport report;
 
     public DocumentValidationException(DocumentValidationReport report) {
-        super();
+        super(HttpServletResponse.SC_PRECONDITION_FAILED);
         this.report = report;
     }
 
+    public DocumentValidationException(String message, int statusCode) {
```

Comment:
Is this constructor really used/needed? It seems we always use `SC_PRECONDITION_FAILED` as `statusCode`.

Reply:
To come back to this comment... after thinking, I think a validation error is *always* `422` (SC_UNPROCESSABLE_ENTITY).
The entity is always readable, but the server refuses because of validation errors.

`411` (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/411) for instance has nothing to do with our validation errors.

Reply:
I agree that `SC_PRECONDITION_FAILED` doesn't seem to me the right error.
Note that CMIS uses `SC_CONFLICT` (409) for most errors having to do with constraints or duplicates. I think it could fit validation errors.

Reply:
FYI CMIS uses `SC_CONFLICT` for `CmisConstraintException`, `CmisContentAlreadyExistsException`, `CmisNameConstraintViolationException`, `CmisUpdateConflictException`, `CmisVersioningException. Cf https://github.com/nuxeo/nuxeo/blob/master/addons/nuxeo-chemistry/nuxeo-opencmis-bindings/src/main/java/org/nuxeo/ecm/core/opencmis/bindings/DefaultErrorExtractor.java#L119

Reply:
And we already use it for `ConcurrentUpdateException` and `DocumentExistsException`

Reply:
@efge  It is acknowledged that `SC_PRECONDITION_FAILED` is not the correct one. Here we are to return `SC_UNPROCESSABLE_ENTITY` (422) in case of validation exception thrown by default (see https://github.com/nuxeo/nuxeo/pull/3489#discussion_r338478093).

Here @troger is rather asking if we want to let the possibility to return something else than `422` in custom code.

Reply:
Ok I see what you mean. Yes you should let the constructor open to providing a different http code, from what you can see above I think it's entirely possible that we'll want custom code to return `409` or even something else.

Reply:
Still not sure a `DocumentValidationException` triggered by our `DocumentValidationService` could be a `409` or something else but well, we can still keep the constructor even if not used.

Reply:
It'd would rather be thrown by custom code by some custom code such an inline core listener on `aboutToCreate` or `beforeDocumentModification` event

Reply:
`DocumentValidationException` seems really tied to `ConstraintViolation` and so on our `DocumentValidationService` ... I would use `NuxeoException` or some other custom Exception for `aboutToCreate` or `beforeDocumentModification` 😛

Reply:
Indeed custom code can throw other exceptions. Let's not complicate for now, just use 422 always then and not make it configurable.

Reply:
Ok